packages/gafl-webapp-service/src/pages/licence-details/licence-length/__tests__/update-transaction.spec.js

```diff
@@ -1,103 +1,133 @@
 import updateTransaction from '../update-transaction'
 import { LICENCE_FULFILMENT } from '../../../../uri.js'
 import * as concessionHelper from '../../../../processors/concession-helper.js'
-import { findPermit } from '../../../../processors/find-permit.js'
+import findPermit from '../../../../processors/find-permit.js'
 import { isPhysical } from '../../../../processors/licence-type-display.js'
 import * as mappings from '../../../../processors/mapping-constants.js'
 import moment from 'moment-timezone'
+import hashPermission from '../../../../processors/hash-permission'
 
 jest.mock('../../../../processors/concession-helper.js')
 jest.mock('../../../../processors/find-permit.js')
 jest.mock('../../../../processors/licence-type-display.js')
+jest.mock('../../../../processors/hash-permission.js', () => jest.fn(() => 'abcde12345'))
 
 describe('licence-details > update-transaction', () => {
   describe('default', () => {
     beforeEach(jest.clearAllMocks)
-    const getMockRequest = (licenceLength = '12M', transactionPageGet = { licenceLength: '' }) => ({
+    const getMockRequest = ({
+      permission = { licenceLength: '12M' },
+      licenceLength = permission.licenceLength,
+      transactionSet = () => {},
+      statusSet = () => {}
+    } = {}) => ({
       cache: jest.fn(() => ({
         helpers: {
           page: {
             getCurrentPermission: () => ({ payload: { 'licence-length': licenceLength } })
           },
           status: {
             getCurrentPermission: () => {},
-            setCurrentPermission: jest.fn()
+            setCurrentPermission: statusSet
           },
           transaction: {
-            getCurrentPermission: () => transactionPageGet,
-            setCurrentPermission: jest.fn()
+            getCurrentPermission: () => permission,
+            setCurrentPermission: transactionSet
           }
         }
       }))
     })
 
     it('should find the permit with the available data', async () => {
```

Comment:
should we test that we are calling hashPermission with permission?

Reply:
Good spot, I'd missed that. I've added it in now